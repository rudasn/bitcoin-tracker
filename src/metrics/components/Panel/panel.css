.panel {
  background-color: #fff;
  border: 1px solid #ccc;
  border-top: 10px solid var(--color-brand);
  box-sizing: border-box;
  margin: var(--gutter) auto var(--gutter);
  max-width: 100%;
  padding: var(--gutter);
  width: var(--width-max);
}
.panel-title {
  margin: 0;
}
.panel-subtitle {
  margin: 0;
}
.panel-chart {
  margin: var(--gutter) 0;
}
.panel-footer {
  text-align: center;
}
