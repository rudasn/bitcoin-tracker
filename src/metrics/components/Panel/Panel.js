import React from 'react'
import propTypes from 'prop-types'

import date from 'utils/date'

import './panel.css'
import PointsTable from '../PointsTable'
import Chart from '../Chart'
import Ticker from '../Ticker'

const Panel = ({ title, value, change=0, percentage=0, dateUpdated=date(), chartOptions={}, tableOptions={}, }) =>
  <section className="panel">
    <header>
      <h2 className="panel-title">{ title }</h2>
      <h3 className="panel-subtitle">
        <Ticker change={ change } value={ value } percentage={ percentage } />
      </h3>
    </header>
    <main>
      <div className="panel-chart">
        <Chart options={ chartOptions } />
      </div>
      <div className="panel-recent-changes">
        <h4>Most Recent Changes</h4>
        <PointsTable rows={ tableOptions.rows } />
      </div>
    </main>
    <footer>
      Last updated: { dateUpdated.toString() }
    </footer>
  </section>

Panel.propTypes = {
  title: propTypes.string,
  value: propTypes.node,
  change: propTypes.number,
  percentage: propTypes.number,
  dateUpdated: propTypes.instanceOf(Date),
  chartOptions: propTypes.object,
  tableOptions: propTypes.object,
}

export default Panel
