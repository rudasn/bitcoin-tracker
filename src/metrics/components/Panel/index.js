import React from 'react'

import './panel.css'
import PointsTable from '../PointsTable'
import Chart from '../Chart'
import Ticker from '../Ticker'

export default ({ title, value, change, percentage, ticker, property, dateUpdated, chartOptions, tableOptions, }) =>
  <section className="panel">
    <header>
      <h2 className="panel-title">{ title }</h2>
      <h3 className="panel-subtitle">
        <Ticker change={ change } value={ value } percentage={ percentage } />
      </h3>
    </header>
    <main>
      <div className="panel-chart">
        <Chart title={ title } options={ chartOptions } />
      </div>
      <div className="panel-recent-changes">
        <h4>Most Recent Changes</h4>
        <PointsTable rows={ tableOptions.rows } />
      </div>
    </main>
    <footer>
      Last updated: { dateUpdated.toString() }
    </footer>
  </section>
