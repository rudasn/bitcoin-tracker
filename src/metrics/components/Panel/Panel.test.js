import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Panel from './Panel'

it('renders without crashing', () => {
  const element = <Panel />
  const div = document.createElement('div')
  ReactDOM.render(element, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders correctly', () => {
  const element = (
    <Panel
      title="This is a panel"
      value={ 1047 }
      change={ 1.05 }
      percentage={ 0.043105 }
    />
  )
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})
