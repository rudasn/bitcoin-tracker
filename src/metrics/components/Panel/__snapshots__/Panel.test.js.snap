// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="panel"
>
  <header>
    <h2
      className="panel-title"
    >
      This is a panel
    </h2>
    <h3
      className="panel-subtitle"
    >
      <div
        className="ticker"
      >
        <span
          className="change-colored-text-up"
        >
          <span
            className="arrow arrow-up"
          />
        </span>
        <span>
          1047
           
        </span>
        <span
          className="change-colored-text-up"
        >
          1.05
           (
          0.04
          %)
        </span>
      </div>
    </h3>
  </header>
  <main>
    <div
      className="panel-chart"
    >
      {
          "options": {}
      }
    </div>
    <div
      className="panel-recent-changes"
    >
      <h4>
        Most Recent Changes
      </h4>
      <table>
        <thead>
          <tr />
        </thead>
        <tbody />
      </table>
    </div>
  </main>
  <footer>
    Last updated: 
    Mon Jan 01 2018 02:00:00 GMT+0200 (Eastern European Standard Time)
  </footer>
</section>
`;
